src/components/ModalContext/ModalTrigger.tsx

```diff
@@ -0,0 +1,19 @@
+import { useModal } from "./ModalHook";
+import { Modal } from "../Modal/Modal";
+interface ModalTriggerArgs {
+  name: string;
+  children: React.ReactNode;
+}
+
+export const ModalTrigger = ({ name, children }: ModalTriggerArgs) => {
+  const { openModal } = useModal(name);
+
+  return (
+    <>
+      <button onClick={openModal}>Open Modal</button>
```

Comment:
I see that there’s no button included in the UI, but the modal functionality looks good to me when playing with the Storybook controls. I would say adding a button for UI testing could be useful but it's not necessary unless there's a specific use for manual button testing.